src/main/java/com/avispl/symphony/dal/infrastructure/management/tripleplay/TriplePlayAggregatorCommunicator.java

```diff
@@ -0,0 +1,601 @@
+	 */
+	private void retrieveInformationOfAllClients(int currentSizeCachedClients) {
+		if (executorService == null || currentSizeCachedClients != cachedClients.size()) {
+			executorService = Executors.newFixedThreadPool(deviceStatisticsCollectionThreads);
+		}
+
+		String lastClientKey = lastClientMac != null ? cachedClients.floorKey(lastClientMac) : null;
+		if (cachedClients.size() == 0) {
+			lastClientMac = null;
+			return;
+		}
+
+		Iterator<Entry<String, Client>> cacheClientIt = cachedClients.entrySet().iterator();
+
+		String itClient = null;
+		if (lastClientKey != null) {
+			while (cacheClientIt.hasNext()) {
+				if (cacheClientIt.next().getKey() == lastClientKey) {
+					break;
+				}
+			}
+		}
+
+		for (int threadNumber = 0; threadNumber < deviceStatisticsCollectionThreads && cacheClientIt.hasNext(); ++threadNumber) {
+			List<String> requestClientMac = new ArrayList<>();
+			for (int clientNumber = 0; clientNumber < TriplePlayConstrant.MAX_CLIENT_QUANTITY_PER_THREAD && cacheClientIt.hasNext(); ++clientNumber) {
+				itClient = cacheClientIt.next().getKey();
+				requestClientMac.add(itClient);
+			}
+			executorService.submit(new ClientLoader(requestClientMac));
+		}
+
+		//Update next ClientIp to next getMultipleStatistics
+		lastClientMac = itClient != cachedClients.lastKey() ? itClient : null;
+	}
+
+	/**
+	 * Retrieve all service
+	 */
+	private void retrieveService() {
+		try {
+			GetAllServicesRequestBody requestBody = new GetAllServicesRequestBody();
+			requestBody.setJsonrpc(TriplePlayConstrant.JSON_RPC);
+			requestBody.setParam(-1);
+			String response = doPost(TriplePlayURL.BASE_URI, requestBody.buildRequestBody());
+			ServiceWrapper monitoringData = objectMapper.readValue(response, ServiceWrapper.class);
+			for (Service service : monitoringData.getServices()) {
+				cachedServices.put(String.valueOf(service.getId()), service);
+			}
+			logger.debug(cachedServices);
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * Get display name of locale
+	 *
+	 * @param locale locale need to get display name, it must follow format locale in response of tripleplay
+	 */
+	private String getDisplayLocalByLocale(String locale) {
+		if (locale.contains(TriplePlayConstrant.SPLIT_LOCALE)) {
+			Locale local = new Locale(locale.substring(0, locale.lastIndexOf(TriplePlayConstrant.SPLIT_LOCALE)), locale.substring(locale.lastIndexOf(TriplePlayConstrant.SPLIT_LOCALE) + 1));
+			return local.getDisplayName();
+		}
+		return null;
+	}
+
+	/**
+	 * Retrieve number of clients and ips of them
+	 */
+	private void retrieveClients() {
+		try {
+			QueryClientsListRequestBody queryClientsListRequest = new QueryClientsListRequestBody();
+			queryClientsListRequest.setJsonrpc(TriplePlayConstrant.JSON_RPC);
+			queryClientsListRequest.setField("clientType");
```

Comment:
refactor code, put all String (clienType, is, ....) to constant

Reply:
I fixed it